ci-operator/step-registry/cucushift/installer/check/vsphere/zone/cucushift-installer-check-vsphere-zone-commands.sh

```diff
@@ -0,0 +1,156 @@
+   fi
+      mapfile -t  node_lists < <(oc get node --no-headers | grep ${node_type} | awk '{print $1}')
+    for node in "${node_lists[@]}"; do
+	    # shellcheck disable=SC2207
+            lable_info=($(oc get node ${node} --show-labels --no-headers | awk -F, '{print $(NF-1),$NF}'))
+            node_region=$(echo ${lable_info[0]} | awk -F= '{print $2}')
+            node_zone=$(echo ${lable_info[1]} | awk -F= '{print $2}')
+        for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+                if [[ ${node_region} == "${zones_region_from_config[${i}]}" && ${node_zone} == "${zones_zone_from_config[${i}]}" ]]; then
+                    if [[ ${node_distribution[*]} =~ ${zones_name_from_config[${i}]} ]]; then
+                            echo "node ${node} located in correct failureDomain. zone name is ${zones_name_from_config[${i}]},region is ${zones_region_from_config[${i}]}, zone is ${zones_zone_from_config[${i}]} " 
+		    else   
+			    check_result=1
+			    echo "node ${node} located in incorrect failureDomain. zone name is ${zones_name_from_config[${i}]},region is ${zones_region_from_config[${i}]}, zone is ${zones_zone_from_config[${i}]} " 
+		    fi
+                fi
+        done
+
+    done
+ 
+return ${check_result}
+}
+
+function check_rhcos() {
+    check_result=0
+    infra_id=$(oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster)
+    # shellcheck disable=SC2207
+    vsphere_datacenters=($(govc ls /))
+    for vsphere_datacenter in "${vsphere_datacenters[@]}"; do
+        folder_name=$(govc ls ${vsphere_datacenter}/vm | grep ${infra_id} )
+	if [[ ${folder_name##*/} == "${infra_id}" ]]; then
+		echo " folder:${folder_name} under vcenter ${vsphere_datacenter} created successful.  "
+	else
+		check_result=1
+		echo " folder name is incorrect. "
+	fi
+    done	    
+
+# check all rhcos images uploaded into each failureDomain.
+        match_rhcos=0
+        for ((i=0;i<${#zones_zone_from_config[@]};i++)); do
+	    rhcos_result=$(govc ls /${zones_datacenter_from_config[${i}]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[${i}]}-${zones_zone_from_config[${i}]} | wc -l)
+            
+	    if [[ ${rhcos_result} == 1 ]]; then
+		    match_rhcos=$(($match_rhcos+1))
+		    echo "rhcos image /${zones_datacenter_from_config[${i}]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[${i}]}-${zones_zone_from_config[${i}]} uploaded to specified failureDomain"
+            else 
+		    echo "rhcos image /${zones_datacenter_from_config[i]}/vm/${infra_id}/${infra_id}-rhcos-${zones_region_from_config[i]}-${zones_zone_from_config[i]} not found in specified failureDomain. please check"
+	            check_result=0
+	    fi
+        done
+        if  [ "${match_rhcos}" -eq "${#zones_zone_from_config[@]}" ]; then
+		echo "all rhcos image was in specified failureDomain"
+        fi
+    
+    
+return ${check_result}    
+}
+
+
+function checkCCM() {
```

Comment:
I have some doubt if we need to check CCM when we already checked node zone setting in `check_node_distribution`.

Reply:
this check point i just followed the case:https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-54872
@jinyunma WDYT?

Reply:
`check_node_distribution` seems only checking the label for each node to make sure that node has correct labels based on setting in install-config (zone/region -- tags on esxi hosts cluster and datacenter).

While cm `cloud-provider-config` contains more platform information, including datacenters, datastore, folders and labels ( tag categories on esxi hosts cluster and datacenter) etc, here make sure that the setting are correct.

Reply:
Per my understanding, the setting in CCM will be passed to vm creation, so I am thinking if we can use `govc` command to check the zone setting is applied the created vms.

Reply:
hmm, I remembered that installer pass 99_openshift-cluster-api_master-machines-x to create each master vm (during testing zonal cluster, I always checked this file when vm creation failed), for cloud-provider-config.yaml, it contains less information, might be used for operators (storage, machine-api etc). 

I agree that using govc command to check if each node created under expected zones, this is one checkpoint described in test case [OCP-54872](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-54872), but missed here.

Reply:
actually my comment is only one principal, this step 's purpose is to test if zonal setting is applied to the cluster, so we do not keep any unrelated things to that. E.g: similar case - `cucushift-installer-check-azure-multi-zones` is doing right things, ensure vms in the cluster are placed into correct zones, nothing else.